sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1325,3 +1327,97 @@ def test_pairwise_boolean_distance():
     nn1 = NN(metric="jaccard", algorithm='brute').fit(X)
     nn2 = NN(metric="jaccard", algorithm='ball_tree').fit(X)
     assert_array_equal(nn1.kneighbors(X)[0], nn2.kneighbors(X)[0])
+
+
+class MyRadiusNNAlgorithm(object):
+    def fit(self, X):
+        self._fitted = True
+
+    def query_radius(self, X, r, return_distance=True):
+        """ Compute radius KNN with brute force"""
+        D = pairwise_distances(X, X)
+        ind = D.argsort(axis=1)
+        D_mask = (D <= r)
+        ind_res = np.empty(D.shape[0], dtype='object')
+        dist = np.empty(D.shape[0], dtype='object')
+        for k in range(D.shape[0]):
+            ind_mask = D_mask[k, ind[k]]
+            ind_res[k] = ind[k, ind_mask]
+            dist[k] = D[k, ind_res[k]]
+
+        if return_distance:
+            return ind, dist
+        return ind
+
+
+class MyKNNAlgorithm(object):
+    def fit(self, X):
+        self._fitted = True
+
+    def query(self, X, k, return_distance=True):
+        """ Compute KNN with brute force"""
+        D = pairwise_distances(X, X)
+        ind = D.argsort(axis=1)
+        ind = ind[:k]
+        dist = D[np.arange(D.shape[0], ind)]
+        if return_distance:
+            return dist, ind
+        return ind
+
+
+@ignore_warnings
+def test_neighbors_estimator_as_algorithm():
+    centers = [[-10, -10], [10, 10]]
+    X, y = make_blobs(centers=centers, random_state=41)
+
+    alg = MyRadiusNNAlgorithm()
+    est = neighbors.NearestNeighbors(n_neighbors=1, algorithm=alg)
+    est.fit([[0], [1]])
+    assert hasattr(alg, '_fitted')
+    for radius in [5, 100]:
+        # check algorithm is queried and parameter is passed
+        dist, ind = est.radius_neighbors([[0]], radius=radius,
+                                         return_distance=True)
+        assert_array_equal(dist[0][0], radius)
```

Comment:
How is it supposed to be equal to `radius`?
Doesn't `dist[0][0]` mean minimum distance between `[[0]]` and `[[0],[1]]` provided `radius` is bigger than distance?

Reply:
That part came from the parent PR.  This was just intended to be a consistency test, I think. If some part of it doesn't make sense to you, you should change it )